src/Estimators/SelfHealingOverlap.cpp

```diff
@@ -78,39 +113,60 @@ void SelfHealingOverlap::accumulate(const RefVector<MCPWalker>& walkers,
     RealType weight        = walker.Weight;
     auto& wcs              = psi.getOrbitals();
 
-    // separate jastrow and fermi wavefunction components
+    // find jastrow wavefunction components
     std::vector<WaveFunctionComponent*> wcs_jastrow;
-    std::vector<WaveFunctionComponent*> wcs_fermi;
     for (auto& wc : wcs)
-      if (wc->isFermionic())
-        wcs_fermi.push_back(wc.get());
-      else
+      if (!wc->isFermionic())
         wcs_jastrow.push_back(wc.get());
 
-    // fermionic must have only one component, and must be multideterminant
-    assert(wcs_fermi.size() == 1);
-    WaveFunctionComponent& wf = *wcs_fermi[0];
-    if (!wf.isMultiDet())
-      throw std::runtime_error("SelfHealingOverlap estimator requires use of multideterminant wavefunction");
-    auto msd_refvec = psi.findMSD();
-    MultiSlaterDetTableMethod& msd = msd_refvec[0];
-
-    // collect parameter derivatives: (dpsi/dc_i)/psi
-    msd.calcIndividualDetRatios(det_ratios);
+    if (wf_type==msd_wf)
+    {
+      auto msd_refvec = psi.findMSD();
+      MultiSlaterDetTableMethod& msd = msd_refvec[0];
+      // collect parameter derivatives: (dpsi/dc_i)/psi
+      if (!use_param_deriv)
+        msd.calcIndividualDetRatios(det_ratios);
+      else
+      {
+        const auto& vars = *msd.myVars;
+        assert(vars.size()>0);
+        msd.evaluateDerivativesWF(pset,vars,det_ratios);
+      }
+    }
+    else if(wf_type==sd_rot_wf)
+    {
+    throw std::runtime_error("SelfHealingOverlap: slaterdet wavefunction implementation incomplete");
+      auto sd_refvec = psi.findSD();
+      //SlaterDet& sd = sd_refvec[0];
+      //if(sd.myVars.size()==0)
+      //  throw std::runtime_error("SelfHealingOverlap: in accumulate, slaterdet has no parameters");
+      //// collect parameter derivatives: (dpsi/dc_i)/psi
+      //sd.evaluateDerivativesWF(pset,sd.myVars,det_ratios);
```

Comment:
Please delete unused code. You can restore them from the history when needed.

Reply:
Done